### PR TITLE
feat: support self-executing Wasm modules

### PR DESCRIPTION
**Pull request by kilyanni**
Closes WARP-77

This adds support for shebang-prefixed Wasm modules so self-executing binaries are recognized and compiled correctly.

Changes:
- add `strip_wasm_shebang` in `wasmer_types`
- strip the shebang before module compilation in the API
- strip the shebang when detecting Wasm binaries on disk in the CLI
- add coverage for module creation, compilation with progress, and extensionless self-executing Wasm files

This makes inputs like `#!/path/to/wasix-run` + Wasm bytes work.

### CODE REVIEW

**Comment by syrusakbary** [comment]
I also added support for shebang as part of my SDK work @kilyanni . 

I believe my commit may be a bit more complete (is also fully tested in some other edge scenarios):
https://github.com/wasmerio/wasmer/pull/6833/changes/a5aa7e2bedf6ebef48f63766dfeb6fba2fadddc4

Feel free to incorporate it to this PR. As long as added tests from my commit are merged and pass on your PR we should be good to go

File: lib/api/src/entities/module/inner.rs [hunk 1]
```diff
@@ -50,6 +51,7 @@ impl BackendModule {
         bytes: impl AsRef<[u8]>,
         callback: CompilationProgressCallback,
     ) -> Result<Self, CompileError> {
+        let bytes = strip_wasm_shebang(bytes.as_ref());
```
Comment (syrusakbary) [inline lib/api/src/entities/module/inner.rs:54]: This is a no-go. Wasmer API should not care at all about shebang (it should be on WASIX - for shebang support inside of WASIX, or CLI for shebang support on the CLI)

File: lib/api/tests/module.rs [hunk 2]
```diff
@@ -32,6 +32,16 @@ fn module_set_name() -> Result<(), String> {
     Ok(())
 }
 
+#[engine_test]
+fn module_new_accepts_shebang_prefixed_wasm() -> Result<(), String> {
```
Comment (syrusakbary) [inline lib/api/tests/module.rs:36]: This should not live here

**Comment by syrusakbary** [comment]
Note: my commit allows running shebang files inside of WASIX, so I think is a different target.

I added a comment on WARP-77.

**Comment by kilyanni** [comment]
Decided in the WASIX sync to close this
